code/game/atoms.dm

```diff
@@ -701,16 +701,56 @@ GLOBAL_LIST_EMPTY(blood_splatter_icons)
 
 
 /mob/living/carbon/human/clean_blood()
-	if(gloves)
-		if(gloves.clean_blood())
-			clean_blood()
-			update_inv_gloves()
-		gloves.germ_level = 0
-	else
-		..() // Clear the Blood_DNA list
-		if(bloody_hands)
+	. = ..()
+	if(head)
+		if(head.clean_blood())
+			update_inv_head()
+	if(wear_suit)
+		if(wear_suit.clean_blood())
+			update_inv_wear_suit()
+	else if(w_uniform)
+		if(w_uniform.clean_blood())
+			update_inv_w_uniform()
+	if(!(wear_suit && wear_suit.flags_inv & HIDEGLOVES))
```

Comment:
I'd much prefer the styling of:

`if(wear_suit && !(wear_suit.flags_inv & HIDEGLOVES))`